### DISCUSSION TITLE
Separated 2D and 3D features for editor

### DISCUSSION THREAD

**Discussion by justlimeguy** [godotengine/godot-proposals#8026] [Engine Core]
I was thinking about it for a while now and recently I found it creating an issue.
I wanted to try the 2d lighting feature but it was limited to Vulkan versions, since not every device has a built in Vulkan this could create problems for people testing/playing the game.
therefore, I am suggesting if the developers could make it so that the engine while creating a projects asks if the user wants to make an entirely 2d or entirely 3d game. and keep the Vulkan features to 3d only, And let the compatibility have the same features as of now.
this could improve the pace ability for users to make efficient games with required features and the engine only load the selected features ( only 2d  or  only 3d  or  general features/compatibility).

**Comment by theraot** [comment]
The compatibility renderer must be completed, so you won't be limited to Vulkan, nor by the compatibility renderer.

About old devices, there has been attempts to bring back a GLES2 for 2D only, but I don't know if that will land on core (I hope it does).

I believe you can disable 3D from editor profiles. I imagine it would be plenty of work to change it so it works per project. Would that really help (considering that you can skip 3D features on 2D projects)?

And you talk about only selected features... Is the Godot editor taking too much to load?

**Reply by justlimeguy** [reply]
i meant like , if a person know the game they are creating is an entirely 2D/ 3D game then they could just choose an 2d/3D editor while creating the game and eliminate the other from the project. So that the editor they are working on will only  have the respective settings/features.
talking about load: yes, sometimes, when i work on a low end device the boot up (even after the first one) takes about a minute to do it.
and sometimes it stops responding and crashes un-intentionally during start.

**Reply by justlimeguy** [reply]
It is just like optimizing engine for the type of game they are creating.
And if they are just testing around or not sure, as you said Compatibility version will be there.
